docs/schema_plugins.md

```diff
@@ -1 +1,234 @@
+
+    x_h5md_creator = SubSection(
+        sub_section=nomad_simulations.schema_packages.general.Program.m_def
+    )
+```
+
+Here, we create a new class called `Simulation()` that inherits from `Simulation()` from `nomad-simulations` while adding the additional `x_h5md_creator` section of type `Program()`. Note that the `x_<program_name>_<quantity_name>` notation is our standard notation for specifying program-specific quantities that are not necessarily generalizable.
+
+Now we can use this new class within our parser to store the creator information:
+
+```python
+from nomad_simulations.schema_packages.general import Program
+from nomad_simulations.schema_packages/<parser_name>_schema.py import Simulation
+
+simulation = Simulation()
+simulation.x_h5md_creator = Program(
+    name=group_h5md_dict.get('h5md_creator_name'),
+    version=group_h5md_dict.get('h5md_creator_version'),
+)
+```
+
+We now need to extend the schema further to store the h5md-nomad author information. For this, let's make a new class with `name` and `email` quantities in `schema_packages/<parser_name>_schema.py`:
+
+```python
+class Author(ArchiveSection):
+    """
+    Contains the specifications of the program.
+    """
+
+    name = Quantity(
+        type=str,
+        shape=[],
+        description="""
+        Specifies the name of the author who generated the h5md file.
+        """,
+    )
+
+    email = Quantity(
+        type=str,
+        shape=[],
+        description="""
+        Author's email.
+        """,
+    )
+```
+
+Again, we need to add this to our custom `Simulation()` class:
+
+ ```python
+class Simulation(nomad_simulations.schema_packages.general.Simulation):
+
+    x_h5md_creator = SubSection(
+        sub_section=nomad_simulations.schema_packages.general.Program.m_def
+    )
+
+    x_h5md_author = SubSection(sub_section=Author.m_def)
+```
+
+And, in our parser:
+
+```python
+from nomad_simulations.schema_packages/<parser_name>_schema.py import Simulation, Author
+
+simulation = Simulation()
+simulation.x_h5md_author = Author(
+    name=group_h5md_dict.get('h5md_author_name'),
+    email=group_h5md_dict.get('h5md_author_email'),
+)
+
+```
+
+Finally, we need to store the h5md-nomad schema version. We could simply add a string quantity to `Simulation()`, however since the h5md schema uses semantic versioning, we could store this as a list of 3 integers to ensure that the user provides the relevant information for this quantity. In `schema_packages/<parser_name>_schema.py`:
+
+
+ ```python
+class Simulation(nomad_simulations.schema_packages.general.Simulation):
+
+    x_h5md_creator = SubSection(
+        sub_section=nomad_simulations.schema_packages.general.Program.m_def
+    )
+
+    x_h5md_author = SubSection(sub_section=Author.m_def)
+
+    x_h5md_version = Quantity(
+        type=np.dtype(np.int32),
+        shape=[3],
+        description="""
+        Specifies the version of the h5md schema being followed.
+        """,
+    )
```

Comment:
Pls mention that this is read from the processed file. My first impression was: "why is this not filled here?"

Btw, this could become a more general quantity, e.g. `format_version`.